osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -211,9 +211,24 @@ protected virtual void UpdateMods()
             if (SelectedItem.Value == null)
                 return;
 
+            ApplyToUserMods();
+
             Mods.Value = UserMods.Value.Concat(SelectedItem.Value.RequiredMods).ToList();
         }
 
+        protected void ApplyToUserMods()
+        {
+            BeatmapDifficulty baseDifficulty = Beatmap.Value.BeatmapInfo.BaseDifficulty;
```

Comment:
Have you tested this when the beatmap is not locally available yet? Basically I think this needs to also be run from `updateWorkingBeatmap`.

Reply:
Before I make this change, can you confirm that we're still wanting to keep this method existing at all, rather than removing it from the mod interface? There are no valid usages of it any more.

Reply:
@peppy I have proposed that above https://github.com/ppy/osu/pull/13825#issuecomment-876699215, with the *conclusion* of keeping the method, but only calling it from `WorkingBeatmap` so that mods that actually require the difficulty values for gameplay purposes or otherwise can still benefit from this feature.

And mods that want this for UI/settings purposes (like Difficulty Adjust) should instead create its own custom control resolving the beatmap bindable.

Alternatively we could just remove that method altogether, but it may be potentially beneficial for some upcoming mod or mods in custom rulesets, as been mentioned during the discussion.

Reply:
I think it's probably okay to remove it... We've already broken compatibility with the DA mod changes, something I forgot about.